### ISSUE TITLE
Adding CPU evaluation support for atrous/dilated convolution?

### ISSUE THREAD

**Issue by DPJep** [Microsoft/CNTK#3137]
Is there a roadmap for adding CPU support for atrous/dilated convolution to CNTK?

**Comment by KeDengMS** [comment]
Yes, we are working with Intel on this, You can take a peek at branch kedeng/mkldnn which had the dilated convolution implementation using MKL-DNN.

**Comment by DPJep** [comment]
Great to hear. I'll check it out.

**Comment by DPJep** [comment]
What is the status / timeline before this is in master?

**Comment by KeDengMS** [comment]
Will be in next release.

**Comment by DPJep** [comment]
Hi @KeDengMS 

I can't find CPU support in ver. 2.6. Is it pushed to 2.7 and what's the ETA?

**Comment by KeDengMS** [comment]
Sorry that it was not in 2.6, as the current focus of CNTK is to export ONNX models. For training models with dilated convolution, please use GPU if possible. For inference, please try export your models with dilated convolution in ONNX format, and run it in [onnxruntime](https://azure.microsoft.com/en-us/blog/onnx-runtime-for-inferencing-machine-learning-models-now-in-preview/). It is a faster, slimmer inference engine than CNTK in lots of ways.

Besides, onnxruntime would soon be open sourced. Keep tuned to our announcements.

**Comment by DPJep** [comment]
Okay, but will it be available at some point in CNTK?
1) so will ONNXruntime also have a C++ API?
2) Will it be both in 32-bit and 64-bit for Windows?
________________________________

**Comment by KeDengMS** [comment]
This would unlikely to be added to CNTK unless there's a strong business reason to support CPU training. The strategy has shifted to focus more on ONNX and inference.
onnxruntime has C/C++/C# APIs, and we will release 64-bit binaries on Windows and Linux. You can also build from source to fit your need. Add @faxu for comments on availability of 32-bit binaries on Windows

**Comment by KeDengMS** [comment]
Yes, CNTK will still be maintained for training.
